torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -730,6 +753,59 @@ void Reducer::finalize_backward() {
     local_used_work_->wait();
   }
   local_used_maps_reduced_ = false;
+
+  // Rebuilt bucket if this is the first time to rebuilt
+  if (!rebuilt_tensors_.empty()) {
+    rebuildBuckets();
```

Comment:
Since we don't need this until the next backward, does it worth it to make the rebuild process async?

Reply:
Since this will launch communication (broadcast), we'd better do it async.

Reply:
Good idea! I think we can make it async for find_unused_parameter = true in the future, as it could rebuild buckets much more times.  For  find_unused_parameter = false, since we only rebuild once, and only broadcast indices not tensors, it should be fast enough.  I will add a TODO comment here. 

How do you think?